src/main/java/org/terasology/launcher/gui/javafx/LogViewController.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.gui.javafx;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+import javafx.concurrent.ScheduledService;
+import javafx.concurrent.Task;
+import javafx.fxml.FXML;
+import javafx.scene.control.TextArea;
+import javafx.scene.text.Font;
+import javafx.util.Duration;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+public class LogViewController extends AppenderBase<ILoggingEvent> {
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS");
+
+    private final StringBuffer buffer;
+
+    @FXML
+    private TextArea logArea;
+
+    public LogViewController() {
+        buffer = new StringBuffer();
+
+        ScheduledService<Void> schedule = new ScheduledService<Void>() {
+            @Override
+            protected Task<Void> createTask() {
+                return new Task<Void>() {
+                    @Override
+                    protected Void call() throws Exception {
+                        logArea.appendText(buffer.toString());
+                        //TODO figure out whether this is thread-safe (I suspect it's not)
+                        buffer.delete(0, buffer.length());
```

Comment:
Another way of clearing the buffer is to set its length to 0. I'm tending towards that, as it is a tad clearer.
```suggestion
                        buffer.setLength(0);
```

Reply:
Due to atomicity of each method the thread might be interrupted and a new log message could have been written to the buffer in the meantime. I don't think it is critical, but the following snippet could solve this problem:

```java
final int length = buffer.length();
logArea.append(buffer.substring(0, length));
buffer.delete(0, length);
```
Not sure whether this is worth the overhead of dealing with substrings and deleting only parts of the buffer.